pyupgrade.py

```diff
@@ -1307,19 +1308,13 @@ def _visit_func(self, node):
             self.generic_visit(node)
 
     def _visit_sync_func(self, node):
-        # type: (Union[ast.FunctionDef, ast.Lambda]) -> None
-        self._in_async_def, orig = False, self._in_async_def
+        # type: (SyncFunctionDef) -> None
+        self._current_sync_def, orig = node, self._current_sync_def
         self._visit_func(node)
-        self._in_async_def = orig
+        self._current_sync_def = orig
 
     visit_FunctionDef = visit_Lambda = _visit_sync_func
 
-    def visit_AsyncFunctionDef(self, node):  # pragma: no cover (py35+)
-        # type: (AsyncFunctionDef) -> None
-        self._in_async_def, orig = True, self._in_async_def
-        self._visit_func(node)
```

Comment:
this introduces a bug, can you write a test for old `super` inside an `async def`?

Reply:
Sorry, I'm afraid I'm not following. Could you elaborate on this?

Reply:
Should I revert back this block and restore `if not self._in_async_def` in `visit_For`?

Reply:
this fails rewrite now:

```python
class C:
    async def foo(self):
        super(C, self).foo()
```

just needs a quick test for that and to make sure this still calls `_visit_func` essentially

Reply:
I see. Thanks for noticing!

Reply:
Addressed in 302ab16283facc67a6cc5ab189af10ba915ae05b.